dask/bag/core.py

```diff
@@ -1134,7 +1134,7 @@ def partition(grouper, sequence, npartitions, p, nelements=2**20):
         d2 = defaultdict(list)
         for k, v in d.items():
             d2[abs(hash(k)) % npartitions].extend(v)
-        p.append(d2)
+        p.append(d2, fsync=True)
```

Comment:
@mrocklin This is where the problem was first spotted: https://gist.github.com/danielballan/3951c1629698601b888c79f49bc837c6. Then @jcrist narrowed the problem down to distributed workers on a single machine, as he described below in this thread.